lib/client.js

```diff
@@ -128,15 +130,186 @@ function sendUncompressedMessage(options, message, logger) {
     performRequest(options, logger);
 }
 
+/**
+ * Puts a metric into the buffer ready to be sent.
+ *
+ * @param self A self statful client.
+ * @param metricTypeConf A configuration for each metric type (counter, gauge, timer). Can be null if it a custom metric.
+ * @param name A metric name.
+ * @param value A metric value.
+ * @param aggregation The aggregation with which metric was aggregated.
+ * @param aggregationFreq The aggregation frequency with which metric was aggregated.
+ * @param parameters An object with metric para meters: tags, agg, aggFreq, namespace and timestamp.
+ */
+function putMetric(self, metricTypeConf, name, value, aggregation, aggregationFreq, parameters, isStats) {
```

Comment:
I can change the behaviour if you want. I chose this way, because the function that the client exposes is strictly the same but we can call it saying it's a stats metrics without having to repeat code..